featuretools/feature_base/features_deserializer.py

```diff
@@ -0,0 +1,112 @@
+            :suppress:
+
+            import featuretools as ft
+            import os
+
+        .. code-block:: python
+
+            filepath = os.path.join('/Home/features/', 'list')
+            ft.load_features(filepath)
+    .. seealso::
+        :func:`.save_features`
+    """
+    return FeaturesDeserializer.load(filepath).to_list()
+
+
+class FeaturesDeserializer(object):
+    FEATURE_CLASSES = {
+        'AggregationFeature': AggregationFeature,
+        'DirectFeature': DirectFeature,
+        'Feature': Feature,
+        'FeatureBase': FeatureBase,
+        'GroupByTransformFeature': GroupByTransformFeature,
+        'IdentityFeature': IdentityFeature,
+        'TransformFeature': TransformFeature,
+    }
+
+    def __init__(self, features_dict):
+        self.features_dict = features_dict
+        self._check_schema_version()
+        self.entityset = deserialize_es(features_dict['entityset'])
+        self._deserialized_features = {}  # name -> feature
+        self._primitives_cache = {}
+
+    @classmethod
+    def load(cls, filepath):
+        with open(filepath, 'r') as f:
+            features_dict = json.load(f)
+
+        return cls(features_dict)
+
+    def to_list(self):
+        feature_names = self.features_dict['feature_list']
+        return [self._deserialize_feature(name) for name in feature_names]
+
+    def _deserialize_feature(self, feature_name):
+        if feature_name in self._deserialized_features:
+            return self._deserialized_features[feature_name]
+
+        feature_dict = self.features_dict['feature_definitions'][feature_name]
+        dependencies_list = feature_dict['dependencies']
+
+        # Collect dependencies into a dictionary of name -> feature.
+        dependencies = {dependency: self._deserialize_feature(dependency)
+                        for dependency in dependencies_list}
+
+        type = feature_dict['type']
+        cls = self.FEATURE_CLASSES.get(type)
+        if not cls:
+            raise RuntimeError('Unrecognized feature type %s' % type)
+
+        args = feature_dict['arguments']
+        feature = cls.from_dictionary(args, self.entityset, dependencies,
+                                      self._primitives_cache)
+
+        self._deserialized_features[feature_name] = feature
+        return feature
+
+    def _check_schema_version(self):
+        current = SCHEMA_VERSION.split('.')
+        saved = self.features_dict['schema_version'].split('.')
+        error_text = ('Unable to load features. The schema version of the saved '
+                      'features (%s) is greater than the latest supported (%s). '
+                      'You may need to upgrade featuretools.'
+                      % (self.features_dict['schema_version'], SCHEMA_VERSION))
+
+        if current[0] < saved[0] or (
+           current[0] == saved[0] and ((current[1] < saved[1]) or (
+                                       current[1] == saved[1] and current[2] < saved[2]))):
```

Comment:
This logic assumes both version numbers have 3 subunits.  We could use a loop:
```python
for c_num, s_num in itertools.zip_longest(current, saved):
    if c_num > s_num:
        break
    elif c_num < s_num:
        raise RuntimeError(error_text)
```

Or add an assert in the test cases that `SCHEMA_VERSION` always has exactly 3 subunits

If we use the loop, itertools calls this `izip_longest` in python2 and `zip_longest` in python3

Reply:
Yeah the loop is easier to read I think. Is the idea to use `zip_longest` and then convert the `None`s to `0`? So that "3.1" < "3.1.1"?

Reply:
Yeah let's use `fillvalue=0` as a kwarg to ziplongest so that "3.1" will be equivalent "3.1.0"